### PR TITLE
Replace XHR request implementation with `fetch`

### PR DESCRIPTION
**Pull request by koddsson**
This means that we have to rely on the [`AbortController`](https://developer.mozilla.org/en-US/docs/Web/API/AbortController) instead of aborting the XHR requests.

Not all browsers support the [`AbortController`](https://developer.mozilla.org/en-US/docs/Web/API/AbortController#Browser_compatibility) so we check before using it. Applications can use a [polyfill](https://github.com/mo/abortcontroller-polyfill) if they want to make sure the element works on more browsers.

This PR doesn't break the API but I want to in the future return the whole response in the error event payload.

#### 📚 References

- [XMLHttpRequest.abort()](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/abort)

### CODE REVIEW

**Comment by koddsson** [comment]
Here's a video of it in action:

Screen Recording 2019-09-23 at 17 31 30

File: src/index.js [hunk 1]
```diff
@@ -117,22 +128,53 @@ function check(autoCheckElement: AutoCheckElement) {
...
+
+  // If there is a controller, it means we are already in flight.
+  // Cancel that request and create a new signal.
+  let controller = abortControllers.get(autoCheckElement)
+  if (controller) {
+    // Cancel the in-flight request.
+    controller.abort()
+  }
+
+  // We need to create a new controller so we can get a new signal?
```
Comment (dgraham) [inline src/index.js]: The previous signal will return `true` in response to `signal.aborted`, so this comment can be removed. It's just how controllers and signals are expected to work.

File: src/index.js [hunk 2]
```diff
@@ -117,22 +128,53 @@ function check(autoCheckElement: AutoCheckElement) {
...
       if (autoCheckElement.required) {
         input.setCustomValidity('')
       }
       input.dispatchEvent(new CustomEvent('auto-check-success', {detail: {message}, bubbles: true}))
+
+      // Mark the component as not being in-flight any more.
+      abortControllers.delete(autoCheckElement)
     })
-    .catch(error => {
+    .catch(async error => {
```
Comment (dgraham) [inline src/index.js]: An `AbortError` is raised when the controller aborts the fetch, which is caught by this block. So we need to ignore aborted fetches here with a [conditional branch](https://developers.google.com/web/updates/2017/09/abortable-fetch#reacting_to_an_aborted_fetch) testing `error.name === 'AbortError'`.

File: src/index.js (hunk 1 above)
Comment (koddsson) [reply src/index.js]: Sounds good! This was a note to my self while I was working on the PR.

**Pull request merged**